gpt_json/gpt.py

```diff
@@ -302,19 +331,46 @@ async def submit_request(
 
         optional_parameters = {}
 
-        if max_response_tokens:
+        if max_response_tokens and self.model in [
+            GPTModelVersion.GPT_3_5,
+            GPTModelVersion.GPT_4,
+        ]:
             optional_parameters["max_tokens"] = max_response_tokens
+        elif self.model in [
+            GPTModelVersion.CLAUD,
+            GPTModelVersion.CLAUD_100K,
+        ]:
+            # need to specify a default for CLAUD models.
+            # using OpenAI's default of 16 tokens: https://platform.openai.com/docs/api-reference/completions/create#completions/create-max_tokens
+            optional_parameters["max_tokens_to_sample"] = (
+                max_response_tokens if max_response_tokens else 16
+            )
 
-        return await openai.ChatCompletion.acreate(
-            model=self.model,
-            messages=[self.message_to_dict(message) for message in messages],
-            temperature=self.temperature,
-            timeout=self.timeout,
-            api_key=self.api_key,
-            stream=stream,
-            **optional_parameters,
-            **self.openai_arguments,
-        )
+        if self.model in [GPTModelVersion.GPT_3_5, GPTModelVersion.GPT_4]:
+            return await openai.ChatCompletion.acreate(
+                model=self.model,
+                messages=[self.message_to_dict(message) for message in messages],
+                temperature=self.temperature,
+                timeout=self.timeout,
+                api_key=self.api_key,
+                stream=stream,
+                **optional_parameters,
+                **self.api_arguments,
+            )
+        elif self.model in [GPTModelVersion.CLAUD, GPTModelVersion.CLAUD_100K]:
+            # format using Claud-mandated user prompt template
+            # see more: https://console.anthropic.com/docs/api/reference#-v1-complete
+            formatted_claud_prompt = (
+                f"{anthropic.HUMAN_PROMPT} {messages[0].content}{anthropic.AI_PROMPT}"
+            )
+            c = anthropic.Client(self.api_key)
+            return await c.acompletion(
+                prompt=formatted_claud_prompt,
+                stop_sequences=[anthropic.HUMAN_PROMPT],
+                model="claude-v1",
```

Comment:
Should this be hard-coded, or tie in with the model version so it can do `claud-v1` vs. `claud-v1-100k`?